plugins/c8y_configuration_plugin/src/main.rs

```diff
@@ -179,12 +196,41 @@ async fn process_mqtt_message(
     mqtt_client: &mut Connection,
     config_file_path: &Path,
     http_client: &mut impl C8YHttpProxy,
+    local_http_host: &str,
     tmp_dir: PathBuf,
+    tedge_device_id: &str,
+    config_dir: &Path,
 ) -> Result<(), anyhow::Error> {
     let health_check_topics = health_check_topics("c8y-configuration-plugin");
-    debug!("Received {:?}", message);
+    let config_snapshot_response: TopicFilter = ConfigOperationTopic::SnapshotResponse.into();
+    let config_update_response: TopicFilter = ConfigOperationTopic::UpdateResponse.into();
+
     if health_check_topics.accept(&message) {
         send_health_status(&mut mqtt_client.published, "c8y-configuration-plugin").await;
+        return Ok(());
+    }
+    if config_snapshot_response.accept(&message) {
+        info!("config snapshot response");
+        let child_config_management = ConfigOperationRequest::try_from(&message)?;
```

Comment:
That doesn't look right. Creating a ConfigOperation**Request** from a config snapshot **response** from the child? Either the enum `ConfigOperationRequest` is named wrong or you're missing another enum definition for `ConfigOperationResponse`.